internal/cypress/grep/parser.go

```diff
@@ -0,0 +1,195 @@
+package grep
+
+import (
+	"regexp"
+	"strings"
+)
+
+// Functions to parse cypress-grep expressions.
+// cypress-grep expressions can include simple logical operations (e.g.
+// NOT, AND, OR). These expressions are parsed into Expressions that are
+// logical predicates that evaluate to true or false depending if the expression matches
+// a given input string.
+
+var reDelimiter = regexp.MustCompile(`[ ,]`)
+
+// Expression is an interface that represents a logical statement.
+type Expression interface {
+	// Eval evaluates the Expression against the input string.
+	Eval(input string) bool
+}
+
+// Partial represents an expression that wraps a string to be matched
+// and evaluates to true if that string is a partial substring of the input.
+type Partial struct {
+	Query  string
+	Invert bool
+}
+
+// Exact represents an expression that wraps a string to be matched
+// and evalutates to true if that string is an exact substring of the input.
+type Exact struct {
+	Query  string
+	Invert bool
+}
+
+// Any represents a composite expression that evaluates to true if any
+// child expression evaluates to true.
+type Any struct {
+	Expressions []Expression
+}
+
+// All represents a composite expression that evaluates to true if all
+// child expressions evaluate to true.
+type All struct {
+	Expressions []Expression
+}
+
+func (p Partial) Eval(input string) bool {
+	contains := strings.Contains(input, p.Query)
+	if p.Invert {
+		return !contains
+	}
+	return contains
+}
+
+func (e Exact) Eval(input string) bool {
+	words := strings.Split(input, " ")
+	contains := false
+
+	for _, w := range words {
+		if w == e.Query {
+			contains = true
+			break
+		}
+	}
+
+	if e.Invert {
+		return !contains
+	}
+	return contains
+}
+
+func (a Any) Eval(input string) bool {
+	for _, p := range a.Expressions {
+		if p.Eval(input) {
+			return true
+		}
+	}
+	return false
+}
+
+func (a *Any) add(p Expression) {
```

Comment:
Go Docs don't actually recommend the mixing of pointers and values in receivers (exceptions exists, e.g. for marshaling). So just add pointers to the receivers of all functions for the structs `All` and `Any`

Reply:
Did not know that. 👍